### PR TITLE
test(mme): Adding unit tests for S1AP MME handler functions

### LINKED ISSUE #9273: Add packet data loading for S1AP handlers unit tests

### PR DESCRIPTION
**Pull request by ardzoht**
Signed-off-by: Alex Rodriguez <ardzoht@gmail.com>

## Summary

- Adds first set of unit tests for S1AP tasks, using test mock ITTI framework introduced on #8487 
    - s1ap_handle_initial_ue_message
    - s1ap_handle_s1_setup_request
    - s1ap_handle_initial_context_setup_response
    - s1ap_handle_ue_context_release_request
    - s1ap_handle_ue_cap_indication
    - added unit tests for s1ap enb stale association
- Adds MME app mock class handler
- Refactoring some S1AP related code 
- Removing unused code and imports

## Test Plan

- make test_oai

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/core/oai/test/s1ap_task/test_s1ap_mme_handlers.cpp [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  ASSERT_EQ(s->num_enbs, 1);
+
+  // Freeing pdu and payload data
+  ASN_STRUCT_FREE_CONTENTS_ONLY(asn_DEF_S1ap_S1AP_PDU, &pdu_s1);
+
+  // Sleep to ensure that messages are received and contexts are released
+  std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+}
+
+TEST_F(S1apMmeHandlersTest, HandleInitialUeMessageSuccess) {
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/test/s1ap_task/test_s1ap_mme_handlers.cpp]: do you plan to add EXPECT_CALLs to the test cases in this PR?

Comment (ardzoht) [reply lte/gateway/c/core/oai/test/s1ap_task/test_s1ap_mme_handlers.cpp]: Added EXPECT_CALLs to the test handlers functions

File: lte/gateway/c/core/oai/test/mock_tasks/sctp_mock.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  return 0;
+}
+
+void stop_mock_sctp_task() {
+  destroy_task_context(&task_zmq_ctx_sctp);
+  pthread_exit(NULL);
+}
+
+void start_mock_sctp_task(std::shared_ptr<MockSctpdHandler> sctpd_handler) {
+  sctpd_handler_ = sctpd_handler;
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/test/mock_tasks/sctp_mock.cpp]: better to use `sctp_handler` than `sctpd_handler` as this is sctp task.

**Review by ulaskozat — APPROVED** [review]
LGTM. Left a minor comment.

**Comment by ulaskozat** [comment]
clang check is failing.

**Pull request merged**